code/modules/mining/machinery/material_extractor.dm

```diff
@@ -1,409 +1,280 @@
-		process_ore(O)
-	else
-		O.dropInto(get_output_loc())
+	var/obj/machinery/atmospherics/portables_connector/port = locate() in loc
+	if(port)
+		var/datum/extension/atmospherics_connection/connection = get_extension(src, /datum/extension/atmospherics_connection)
+		if(connection)
+			connection.connect(port)
 
-/obj/machinery/atmospherics/unary/material/extractor/ProcessAtomTemperature()
-	if(operable() && use_power)
-		//We process temp for the input reagent_holder too
-		input_buffer.temperature = GAS_EXTRACTOR_OPERATING_TEMP
-		temperature = GAS_EXTRACTOR_OPERATING_TEMP
-		return TRUE
-	return ..()
 
-/obj/machinery/atmospherics/unary/material/extractor/examine(var/mob/user)
+/obj/machinery/material_processing/extractor/examine(mob/user)
 	. = ..()
-	//Only display info if the screen is there
-	if(get_component_of_type(/obj/item/stock_parts/console_screen))
-		to_chat(user, SPAN_NOTICE("The processing tank gauge reads [round(input_buffer.reagents.total_volume)]/[input_buffer.reagents.maximum_volume] units of liquid."))
-		to_chat(user, SPAN_NOTICE("The internal storage tank gauge reads [round(reagents.total_volume)]/[reagents.maximum_volume] units of liquid."))
-		to_chat(user, SPAN_NOTICE("The internal gas tank pressure gauge reads [air_contents.return_pressure()] kPa."))
-
-		if(is_output_container_full() || is_internal_tank_full())
-			to_chat(user, SPAN_WARNING("It is currently idling because one or more of its liquid tanks are full."))
-		else
-			to_chat(user, SPAN_NOTICE("Everything is working correctly."))
 
-	if(output_container)
-		var/output_desc = SPAN_NOTICE("It has \a [output_container.name] in place to receive reagents.")
-		if(is_output_container_full())
-			output_desc = "[output_desc] [SPAN_WARNING("It's full!")]"
-		to_chat(user, output_desc)
+	var/datum/extension/atmospherics_connection/connection = get_extension(src, /datum/extension/atmospherics_connection)
+	if(connection.connected_port)
+		to_chat(user, SPAN_NOTICE("It is connected to \the [connection.connected_port]."))
 	else
-		to_chat(user, SPAN_NOTICE("It has nothing to pour reagents into."))
+		to_chat(user, SPAN_NOTICE("It may be connected to an atmospherics connector port with a wrench."))
+
+	if(atom_flags & ATOM_FLAG_OPEN_CONTAINER)
+		to_chat(user, SPAN_NOTICE("Its open for refilling."))
+
+/obj/machinery/material_processing/extractor/Process()
+	if(!use_power || (stat & (BROKEN|NOPOWER)))
+		return
+
+	if(reagents?.total_volume >= max_liquid)
+		return
+
+	if(input_turf)
+		var/eaten = 0
+		for(var/obj/item/eating in input_turf)
+			if(!eating.simulated || eating.anchored)
+				continue
+			if(!can_eat(eating))
+				if(output_turf)
+					eating.dropInto(output_turf)
+				continue
+			eaten++
+			if(eating.reagents?.total_volume)
+				eating.reagents.trans_to_obj(src, eating.reagents.total_volume)
+			for(var/mtype in eating.matter)
+				reagents.add_reagent(mtype, FLOOR(eating.matter[mtype] * REAGENT_UNITS_PER_MATERIAL_UNIT))
```

Comment:
Setting the noreact flag will disable heating products as well, so that's a no-go as it currently is. I can change that, but I don't actually think there's a problem. You need to be careful with what you put into the extractor so that you don't get unintended reactions. If specific ices have heating products that immediately react, then that's a problem with the ices' products, and not the machinery in my opinion.

Reply:
> Setting the noreact flag will disable heating products as well, so that's a no-go as it currently is. I can change that, but I don't actually think there's a problem. You need to be careful with what you put into the extractor so that you don't get unintended reactions. If specific ices have heating products that immediately react, then that's a problem with the ices' products, and not the machinery in my opinion.

Yes, but that was the main issue with the previous iteration. There's always going to be a possibility for reactions currently no matter what you put in. 
Especially since people may add more reactions in the future. So you'd have to constantly change things each times a recipe might happen with the ices you allow.. Which is a lot of work, and essentially repeating the mistakes I made.

Also, consider that, if you can't actually extract those things separately from ices, this machine is essentially useless for reagents. You can just use a reagent grinder since it'll do the same thing.

Reply:
Fair enough. I've split up the atom flags and set the extractor to only perform phase changes, not chemical reactions or dissolving.